doozer/doozerlib/cli/scan_osh.py

```diff
@@ -0,0 +1,179 @@
+        """
+        Returns the latest RPMs and builds tagged in to the candidate tag received as input
+        """
+        latest_tagged = self.koji_session.listTagged(tag=tag, latest=True)
+        if latest_tagged:
+            return latest_tagged
+        else:
+            return []
+
+    def get_tagged_all(self, tag):
+        """
+        Returns all the RPMs and builds that are currently in the candidate tag received as input
+        """
+        latest_tagged = self.koji_session.listTagged(tag=tag, latest=False)
+        if latest_tagged:
+            return latest_tagged
+        else:
+            return []
+
+    def trigger_scans(self, nvrs: list):
+        cmd_template = "osh-cli mock-build --config={config} --brew-build {nvr} --nowait"
+        for nvr in nvrs:
+            # Skip rhcos for now
+            if nvr.startswith("rhcos"):
+                self.runtime.logger.warning(f"Skipping RHCOS builds. Scan is not triggered for {nvr}")
+                continue
+
+            if "container" in nvr:
+                cmd = cmd_template.format(config="cspodman", nvr=nvr)
+
+            else:
+                if "el7" in nvr:
+                    rhel_version = 7
+                elif "el8" in nvr:
+                    rhel_version = 8
+                elif "el9" in nvr or nvr.startswith("rhcos"):
+                    rhel_version = 9
+                else:
+                    self.runtime.logger.error("Invalid RHEL version")
+                    return
+
+                cmd = cmd_template.format(config=f"rhel-{rhel_version}-x86_64", nvr=nvr)
+
+            message = f"Ran command: {cmd}"
+
+            if not self.dry_run:
+                fire_and_forget(self.runtime.cwd, cmd)
+            else:
+                message = "[DRY RUN] " + message
+
+            self.runtime.logger.info(message)
+
+        self.runtime.logger.info(f"Total number of build scans kicked off: {len(nvrs)}")
+
+    async def run(self):
+        if not self.specific_nvrs:
+            tags = self.runtime.get_errata_config()["brew_tag_product_version_mapping"].keys()
+            for tag in tags:
+                major, minor = self.runtime.get_major_minor_fields()
+                self.brew_tags.append(tag.format(MAJOR=major, MINOR=minor))
+
+            builds = []
+
+            if self.last_brew_event or self.all_builds:
+                for tag in self.brew_tags:
+                    builds += self.get_tagged_all(tag=tag)
+
+                if self.last_brew_event:
+                    builds = [build for build in builds if build["create_event"] > self.last_brew_event]
+            else:
+                # If no --since field is specified, find all the builds that have been tagged into our candidate tags
+                for tag in self.brew_tags:
+                    builds += self.get_tagged_latest(tag=tag)
+
+            # Sort the builds based on the event ID by descending order so that latest is always on top
```

Comment:
also mention that create_Event is the event on which the build was tagged in the tag and not the build creation time